classes/services/PKPContextService.inc.php

```diff
@@ -299,6 +299,37 @@ public function validate($action, $props, $allowedLocales, $primaryLocale) {
 			}
 		});
 
+		// Ensure that the form and submission locales are available for the site
+		$validator->after(function($validator) use ($action, $props) {
+			$getCurrentPropValue = function($prop) use ($props) {
+				if (isset($props[$prop])) {
+					return $props[$prop];
+				}
+
+				$context = Application::get()->getRequest()->getContext();
+				if (isset($props['id']) && $context->getId() !== $props['id']) {
+					$context = $this->get($props['id']);
+				}
+
+				return $context->getData($prop);
+			};
+
+			$supportedLocales = $getCurrentPropValue('supportedLocales');
+			$supportedFormLocales = $getCurrentPropValue('supportedFormLocales');
+			$supportedSubmissionLocales = $getCurrentPropValue('supportedSubmissionLocales');
+
+			$unsupportedFormLocales = array_diff($supportedFormLocales, $supportedLocales);
+			$unsupportedSubmissionLocales = array_diff($supportedSubmissionLocales, $supportedLocales);
+
+			if ($unsupportedFormLocales !== [] && !$validator->errors()->has('supportedFormLocales')) {
```

Comment:
I think that we need to only throw these validation errors if a prop is passed. So only run these checks if `isset($props['supportedFormLocales'])`. That will allow a situation where the `supportedLocales` are changed and we automatically remove the `supportedFormLocales`.

The same is true for `supportedSubmissionLocales` below.